pkg/store/findings.go

```diff
@@ -718,11 +718,22 @@ func replaceFindingsState(ctx context.Context, tx *sqlx.Tx, l *log.Logger, state
 // sanitizeJSONB sanitizes a JSON string to comply with PostgreSQL
 // restrictions for unicode escape sequences, particularly for the
 // NULL character (\u0000).
+// Just removing \u0000 can generate a wrong string in this case \\u0000 (\).
 // See doc: https://www.postgresql.org/docs/10/datatype-json.html.
 func sanitizeJSONB(jsonb *[]byte) *[]byte {
 	if jsonb == nil {
 		return jsonb
 	}
-	sanitized := []byte(strings.Replace(string(*jsonb), "\\u0000", "", -1))
+	// "\\u0000" "\\\\u0000" are correct
+	// "\u0000" "\\\u0000" are not
+	s := string(*jsonb)
+	r := regexp.MustCompile(`((^|[^\\])(\\\\)*)(\\u0000)+`)
```

Comment:
Not a big deal, but we usually compile the regex outside the function in a var, so it is cached and we don't need to compile it each time the function is executed. So putting something like this as global var at the beginning of the file:
```go
var nullCharRegExp = regexp.MustCompile(`((^|[^\\])(\\\\)*)(\\u0000)+`)
``` 
and then just reference that variable here, so something like that:
```go
 nullCharRegExp.ReplaceAllString(s, "$1")
```

Reply:
that makes a lot of sense. thanks